src/test/java/com/scalar/db/storage/dynamo/ScannerImplTest.java

```diff
@@ -0,0 +1,166 @@
+package com.scalar.db.storage.dynamo;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.scalar.db.api.Result;
+import com.scalar.db.api.Scan;
+import com.scalar.db.io.IntValue;
+import com.scalar.db.io.Key;
+import com.scalar.db.io.TextValue;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+public class ScannerImplTest {
```

Comment:
I think `ScannerImplTest` should test if `isSingleClusteringKey` check is properly working and the expected behaviors are happening for each case.
For example, if `isSingleClusteringKey` is `false` then checking if the sorting function is called.
The sorting function itself should be tested in the unit test of `ItemSorter` (a sorting class whatever the name is).

Reply:
Thank you! I've added `ItemSorterTest`. Could you take a look again?